frontend/src/components/common/Resource/EditorDialog.tsx

```diff
@@ -190,7 +194,7 @@ export default function EditorDialog(props: EditorDialogProps) {
     if (!format || (!res.obj && looksLikeJson(code))) {
       res.format = 'json';
       try {
-        res.obj = JSON.parse(code) as KubeObjectInterface;
+        res.obj = JSON.parse(code) as KubeObjectInterface[];
```

Comment:
This will mask the case when the object is a single resource and not multiple.
Also, have you checked if kubectl accepts multiple resources as JSON?